developer-docs-site/docs/standards/object.md

```diff
@@ -0,0 +1,239 @@
+4. Data cannot be recursively composable, because Move currently prohibits recursive data structures. Furthermore, experience suggests that true recursive data structures can lead to security vulnerabilities.
+5. Existing data cannot be easily referenced from entry functions, for example, supporting string validation requires many lines of code. Each key within a table can be very unique and specializing to support within entry functions becomes complex.
+6. Events cannot be emitted from data but from an account that may not be associated with the data.
+7. Transferring logic is limited to the APIs provided in the respective modules and generally requires loading resources on both the sender and receiver adding unnecessary cost overheads.
+
+:::tip
+Object is a core primitive in Aptos Move and created via the object module at 0x1::object
+:::
+
+## Structure
+An object is stored in the ObjectGroup resource group, which enables other resources within the object to be co-located for data locality and data cost savings. It's important to note that not all resources within an object need to be co-located within the ObjectGroup, and it's up to the developer of an object to determine their data layout.
+
+[object.png](../../static/img/docs/object.png)
+
+### Object resource group
+
+Object is a container for resources that are stored within a single address. These resources usually represent related data often accessed together and should be stored within a single address for data locality and cost savings.
+When created, an object has a resource group, ObjectGroup, by default:
+```rust
+#[resource_group(scope = global)]
+struct ObjectGroup { }
+```
+
+Each object also has the core ObjectCore resource with fundamental properties:
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+struct ObjectCore has key {
+    /// Used by guid to guarantee globally unique objects and create event streams
+    guid_creation_num: u64,
+    /// The address (object or account) that owns this object
+    owner: address,
+    /// Object transferring is a common operation, this allows for disabling and enabling
+    /// transfers. Bypassing the use of a the TransferRef.
+    allow_ungated_transfer: bool,
+    /// Emitted events upon transferring of ownership.
+    transfer_events: event::EventHandle<TransferEvent>,
+}
+```
+
+After creating an object, creators can extend with additional resources. For example, an exchange can create an object for each of its liquidity pools and add a resource to track the pool's liquidity.
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+struct LiquidityPool has key {
+    token_a: Object<FungibleAsset>,
+    token_b: Object<FungibleAsset>,
+    reserves_a: u128,
+    reserves_b: u128
+}
+```
+
+token_a and token_b are references to other objects. In this example, they're fungible assets (similar to coins). This will be covered in more detail in the fungible asset standard.
```

Comment:
```suggestion
In the above code, `token_a` and `token_b` are references to other objects. Specifically, `Object<T>` is a reference to an object stored at a given address that contains `T` resource. In this example, they're fungible assets (similar to coins). This will be covered in more detail in the fungible asset standard.
```